src/routes/targeted-messaging/targeted-messaging.service.ts

```diff
@@ -0,0 +1,79 @@
+import { SubmissionNotFoundError } from '@/domain/targeted-messaging/errors/submission-not-found.error';
+import { ITargetedMessagingRepository } from '@/domain/targeted-messaging/targeted-messaging.repository.interface';
+import { CreateSubmissionDto } from '@/routes/targeted-messaging/entities/create-submission.dto.entity';
+import { Submission } from '@/routes/targeted-messaging/entities/submission.entity';
+import { BadRequestException, Inject, Injectable } from '@nestjs/common';
+
+@Injectable()
+export class TargetedMessagingService {
+  constructor(
+    @Inject(ITargetedMessagingRepository)
+    private readonly repository: ITargetedMessagingRepository,
+  ) {}
+
+  async getSubmission(args: {
+    outreachId: number;
+    chainId: string;
+    safeAddress: `0x${string}`;
+    signerAddress: `0x${string}`;
+  }): Promise<Submission> {
+    const targetedSafe = await this.repository.getTargetedSafe(args);
+    try {
+      const submission = await this.repository.getSubmission({
+        chainId: args.chainId,
+        targetedSafe,
+        signerAddress: args.signerAddress,
+      });
+      return new Submission(
+        submission.outreachId,
+        submission.targetedSafeId,
```

Comment:
This continues from my `outreachId` comment, the `targetedSafeId` is effectively the Safe address. Do we want to/need to expose this?

This can be tackled in a separate PR if we think necessary.

Reply:
I've answered in the other comment, thanks for bringing this up!